src/main/attached_database.cpp

```diff
@@ -257,39 +277,28 @@ void AttachedDatabase::OnDetach(ClientContext &context) {
 	}
 }
 
-void AttachedDatabase::Close() {
-	if (is_closed) {
+void AttachedDatabase::Checkpoint() {
```

Comment:
Are we ever calling `Checkpoint()` without closing the database?

Given that `Checkpoint` can also only ever be called once due to the `checkpoint` flag being set - I think this is a bit of a misnomer.

Maybe we can parameterize the `Close` instead, e.g.:

```cpp
Close(DatabaseCloseAction::CHECKPOINT); // throws if checkpoint fails, but still cleans up regardless
Close(DatabaseCloseAction::TRY_CHECKPOINT); // tries to checkpoint, if it fails does not throw
Close(DatabaseCloseAction::CLEANUP_ONLY); // does not try to checkpoint

```

Reply:
I like this! Will make the change!